core/model/modx/modfilehandler.class.php

```diff
@@ -613,4 +602,101 @@ public function remove($options = array()) {
+     * @option string|array extensions Comma separated list or array of file extensions to filter files by
+     * @option boolean|function callback Anonymous function to modify each output item, $item will be passed / accessible
+     *      
+     * @return array
+     */
+    public function getList($options = array()) {
+        $options = array_merge(array(
+            'recursive' => false,
+            'sort' => false,
+            'sortdir' => SORT_ASC,
+            'sortflag' => SORT_REGULAR,
+            'skiphidden' => true,
+            'skipdirs' => true,
+            'skip' => array(),
+            'extensions' => array(),
+            'callback' => false,
+        ), $options);
+
+        $items = array();
+
+        $mb = $this->fileHandler->modx->getOption('use_multibyte', null, false);
+        $mbencoding = $this->fileHandler->modx->getOption('modx_charset', null, 'UTF-8');
+        $extensions = !is_array($options['extensions']) ? explode(',', $options['extensions']) : $options['extensions'];
+        $skip = !is_array($options['skip']) ? explode(',', $options['skip']) : $options['skip'];
+        $iterator = $options['recursive'] ? new RecursiveIteratorIterator(new RecursiveDirectoryIterator($this->path)) : new DirectoryIterator($this->path);
+
+        foreach ($iterator as $item) {
+            $skipfile = !empty($skip) ? in_array($item->getFilename(), $skip) : false;
+            $ishidden = false;
+
+            if ($options['skiphidden']) {
+                // check for hidden folder, also hide with visible ones inside
+                // but don't skip weird filenames like "...and-there-was-silence.avi"
+                if ($item->isDot() || preg_match('/(\/\.\w+|\\\.\w+)/', $item->getPath())) {
+                    continue;
+                }
+                // check for hidden file (probably works only on UNIX filesystems)
+                $ishidden = preg_match('/^(\.\w+)/i', $item->getFilename());
+            } else if (!$options['skipdirs']) {
+                // always exclude . and .. directory navigators, only relevant when including folders
+                $ishidden = $item->isDot();
+            }
+
+            if (($item->isFile() || $item->isDir() && !$options['skipdirs']) && !$ishidden && !$skipfile) {
+                $additem = true;
+                
+                if (!empty($options['extensions'])) {
+                    // if min PHP version is 5.3.6 we can use $item->getExtension()
+                    $extension = pathinfo($item->getPathname(), PATHINFO_EXTENSION);
+                    $extension = $mb ? mb_strtolower($extension, $mbencoding) : strtolower($extension);
+
+                    if (!in_array($extension, $extensions)) {
+                        $additem = false;
+                    }
+                }
+
+                if ($additem) {
+                    if (is_object($options['callback']) && ($options['callback'] instanceof Closure)) {
```

Comment:
Well, i guess we would then be allowed to use [call_user_func](http://php.net/manual/en/function.call-user-func.php) (or call_user_func_array), allowing us to use more than just clojure : `$items[] = call_user_func($options['callback'], $item)`.